src/components/pages/deploy/ErrorMessagesFragment.tsx

```diff
@@ -13,20 +12,34 @@ const Wrapper = styled.div`
 `;
 
 function component(): JSX.Element {
-  const messages = useRecoilValue(messagesSelector);
+  const { errors = {}, touched = {}, submitErrors = {} } = useFormState({
+    subscription: { errors: true, touched: true, submitErrors: true },
+  });
 
-  if (!messages) {
+  if ((!errors || !Object.values(touched).some(Boolean)) && !submitErrors) {
```

Comment:
How can you use `Boolean` as the check-function here? Does it convert the `touched` entries into booleans?

Reply:
`touched` is an array of booleans already https://final-form.org/docs/final-form/types/FormState#touched

Even if values weren't boolean, the expression would just filters out whatever is falsy.

In summary, In case there's no field that's touched, I don't want to show errors.

```
[false, false].some(Boolean) // returns false, negated => true
[true, false].some(Boolean) // returns true, negated => false
```

As for `Boolean`, this explains it https://stackoverflow.com/a/54623591/1272513